tests/ui/let_unit.fixed

```diff
@@ -15,16 +15,25 @@ fn main() {
     let _y = 1; // this is fine
     let _z = ((), 1); // this as well
     if true {
-        ();
+        // do not lint this, since () is explicit
+        let _a = ();
     }
 
+    // do not lint this, since () is explicit
+    let () = dummy();
+    let () = ();
+    () = dummy();
+    () = ();
+
```

Comment:
If we could cover the following as well it would be quite neat:

```suggestion
let _: () = dummy();

```

Reply:
Fine by me